include/gzip/config.hpp

```diff
@@ -0,0 +1,5 @@
+#pragma once
```

Comment:
What is the vision  of this file?

Reply:
One single place to define `ZLIB_CONST` (without this it would need to be defined in every header before `zlib.h` is included. This consolidates enabling the `const` type inside zlib in one place.